Gems/ONNX/Code/Source/Clients/Model.cpp

```diff
@@ -12,10 +12,27 @@ namespace ONNX
 {
     void Model::Load(const InitSettings& initSettings)
     {
+        AZStd::wstring modelFile;
+
+        // If no filepath provided for onnx model, set default to a model.onnx file in the Assets folder.
+        if (initSettings.m_modelFile.empty())
+        {
+            // Get the FileIOBase to resolve the path to the ONNX gem
+            AZ::IO::FileIOBase* fileIo = AZ::IO::FileIOBase::GetInstance();
+            AZ::IO::FixedMaxPath onnxGemRoot;
+            fileIo->ResolvePath(onnxGemRoot, "@gemroot:ONNX@");
+            AZStd::string gemRootString = onnxGemRoot.c_str();
+            modelFile = AZStd::wstring(gemRootString.begin(), gemRootString.end()) + AZStd::wstring(L"/Assets/model.onnx");
```

Comment:
You should  use the PathAppend operator before converting it to a wstring to reduce the string operations

```suggestion
            onnxGemRoot /= "/Assets/model.onnx";
            modelFile = AZStd::to_wstring(onnxGemRoot.Native());;
```

Reply:
As a matter of fact there is isn't a need to convert to a wstring until the all to Ort::Session::Session.
Until then the string should be kept as a path.

Reply:
Changed